x-pack/plugin/esql/compute/src/main/java/org/elasticsearch/compute/aggregation/X-ValueByTimestampAggregator.java.st

```diff
@@ -145,7 +184,14 @@ public class $Occurrence$$Type$ByTimestampAggregator {
             }
             if (updated) {
                 values = bigArrays.grow(values, groupId + 1);
+                $if(BytesRef)$
+                try(BreakingBytesRefBuilder builder = new BreakingBytesRefBuilder(breaker, "$Occurrence$", value.length)) {
```

Comment:
You *can't* put this in the `try` or we'll release the tracking as soon as the block is done. Instead we need to release all of the elements in `close`

Reply:
Instead of making a new `BreakingBytesRefBuilder` we should fill the `values` array as we expand it. So we can call `values.get(groupId).copyBytes(value)` every time. We *mostly* don't expand the array for unused groups.

Reply:
I applied your suggestion as I understood! Please let me know if the manual creation + release is not what you meant